### PR TITLE
Support account "bell" notifications #1961

### PR DESCRIPTION
**Pull request by Oreolek**
You can subscribe to any account you're following.

### CODE REVIEW

File: src/routes/_components/profile/AccountProfileNotify.html [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      },
+      async onNotifyButtonClick (e) {
+        e.preventDefault()
+        e.stopPropagation()
+        const {
+          account,
+          accountId,
+          notifying
+        } = this.get()
+        if (notifying) { // unblock
```
Comment (nolanlawson) [inline src/routes/_components/profile/AccountProfileNotify.html:56]:
```suggestion
        if (notifying) { // notify
```

File: src/routes/_components/profile/AccountProfileNotify.html [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          await setAccountNotify(accountId, false)
+        } else { // follow/unfollow
```
Comment (nolanlawson) [inline src/routes/_components/profile/AccountProfileNotify.html:58]:
```suggestion
        } else { // notify
```

File: src/routes/_components/profile/AccountProfileNotify.html [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      accountId: ({ account }) => account.id,
+      notifying: ({ relationship }) => {
+        return relationship && relationship.notifying
+      },
+      href: ({ notifying }) => {
+        if (notifying) {
+          return '#fa-bell-ringing'
+        }
+        return '#fa-bell-o'
+      },
```
Comment (nolanlawson) [inline src/routes/_components/profile/AccountProfileNotify.html:77]:
```suggestion
      href: ({ notifying }) => {
        notifying ? '#fa-bell-ringing' : '#fa-bell-o'
      ),
```

File: src/routes/_components/profile/AccountProfileNotify.html [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      label: ({ notifying }) => {
+        if (notifying) {
+          return formatIntl('intl.notifyLabel')
+        }
+        return formatIntl('intl.denotifyLabel')
+      },
```
Comment (nolanlawson) [inline src/routes/_components/profile/AccountProfileNotify.html:83]:
```suggestion
      label: ({ notifying }) => (
        notifying ? formatIntl('intl.notifyLabel') : formatIntl('intl.denotifyLabel')
      ),
```

File: src/routes/_components/profile/AccountProfileNotify.html [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    data: () => ({
+    }),
+    computed: {
+      accountId: ({ account }) => account.id,
+      notifying: ({ relationship }) => {
+        return relationship && relationship.notifying
+      },
+      href: ({ notifying }) => {
+        if (notifying) {
+          return '#fa-bell-ringing'
```
Comment (nolanlawson) [inline src/routes/_components/profile/AccountProfileNotify.html:74]: There is no fa-bell-ringing, at least in the version of fontawesome I'm using. Did you mean just "fa-bell"?

File: src/routes/_components/profile/AccountProfileNotify.html [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  import { setAccountNotify } from '../../_actions/notify.js'
+  import { formatIntl } from '../../_utils/formatIntl.js'
+
+  export default {
+    methods: {
+      oncreate () {
+        if (process.browser) {
+          window.__button = this
+        }
+      },
```
[the remark this answers is not in the document]
Comment (nolanlawson) [reply src/routes/_components/profile/AccountProfileNotify.html]: removed

**Comment by nolanlawson** [comment]
I have a few quibbles with the style of the bell icon:

Screenshot from 2022-04-28 08-20-55
Screenshot from 2022-04-28 08-20-50

- When "on", the color doesn't match the same "on" color for the "following" button
- When toggling following on/off, the button disappears, causing content to shift around slightly
- It seems visually noisy to me to have this extra button, which I'm not sure is a super-popular feature

How would you feel about putting the notify/de-notify option behind the hamburger (≡) button? Then we wouldn't need the bell at all.

File: src/routes/_components/profile/AccountProfileNotify.html (hunk 5 above)
Comment (Oreolek) [reply src/routes/_components/profile/AccountProfileNotify.html]: i used [fork-awesome](https://forkaweso.me/Fork-Awesome/icons/) as the iconsheet. I find the icons `fa-bell` and `fa-bell-o` too similar, especially in some color schemes. It's not apparent in the Mastodon-FE interface if the bell is just an outline on transparent background if the background is dark. The newer FA versions have a `fa-bell-ringing` icon that can show the notification status more clearly.

**Comment by Oreolek** [comment]
It is near the follow button in Mastodon-FE interface, that's why I put it there. As much as "follow and subscribe" is a natural combo, the menu line **would** be better because it would solve the ambiguous icon problem.

**Comment by nolanlawson** [comment]
This needed a bit more work, like:

- tests
- showing a different icon/text in the notification
- making it work with notification filters
- making push notifications work

This should work, although I haven't tested push notifications.

**Comment by nolanlawson** [comment]
Push notifications work, let's do it. Thanks for the PR!

**Pull request merged**